runtime/sam/expr/function/time.go

```diff
@@ -21,30 +21,36 @@ type Bucket struct {
 }
 
 func (b *Bucket) Call(_ super.Allocator, args []super.Value) super.Value {
+	args = underAll(args)
 	tsArg := args[0]
 	binArg := args[1]
 	if tsArg.IsNull() || binArg.IsNull() {
+		if tsArg.Type().ID() == super.IDDuration {
+			return super.NullDuration
+		}
 		return super.NullTime
 	}
-	var bin nano.Duration
-	if binArg.Type() == super.TypeDuration {
-		bin = nano.Duration(binArg.Int())
-	} else {
-		d, ok := coerce.ToInt(binArg, super.TypeDuration)
-		if !ok {
-			return b.zctx.WrapError(b.name+": second argument is not a duration or number", binArg)
-		}
-		bin = nano.Duration(d) * nano.Second
+	i, ok := coerce.ToInt(binArg, super.TypeInt64)
+	if !ok {
+		return b.zctx.WrapError(b.name+": second argument is not a duration or number", binArg)
```

Comment:
Nit: This error string isn't accurate since coerce.ToInt can succeed when the statement isn't true and fail when it is true.